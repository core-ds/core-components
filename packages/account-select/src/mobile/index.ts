export { AccountSelectMobile } from './Component.mobile';
