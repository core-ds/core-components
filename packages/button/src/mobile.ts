export { ButtonMobile } from './Component.mobile';
export type { CommonButtonProps as ButtonMobileProps } from './typings';
