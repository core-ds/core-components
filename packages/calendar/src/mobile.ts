export * from './components/calendar-mobile';
