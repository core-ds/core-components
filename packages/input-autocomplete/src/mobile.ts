export * from './Component.mobile';
export * from './Component.modal.mobile';
