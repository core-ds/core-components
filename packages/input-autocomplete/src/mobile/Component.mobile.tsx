import React, { Ref, useMemo, useRef, useState } from 'react';
import mergeRefs from 'react-merge-refs';
import cn from 'classnames';
import throttle from 'lodash/throttle';

import {
    SelectMobile,
    SelectMobileProps,
    SelectModalMobile,
} from '@alfalab/core-components-select/mobile';
import {
    AnyObject,
    BottomSheetSelectMobileProps,
    Footer,
    ModalSelectMobileProps,
} from '@alfalab/core-components-select/shared';

import { AutocompleteMobileField } from '../autocomplete-mobile-field';
import { OnInputReason } from '../enums';
import { InputAutocompleteMobileProps } from '../types';
import { searchFilterStub } from '../utils';

import styles from './mobile.module.css';

export const InputAutocompleteMobile = React.forwardRef(
    (
        {
            Input,
            value,
            name,
            Arrow = null,
            label,
            placeholder = '',
            size = 48,
            open: openProp,
            onInput,
            onOpen,
            multiple,
            inputProps,
            isBottomSheet = true,
            dataTestId,
            transitionProps,
            onCancel,
            onApply,
            title,
            success,
            ...restProps
        }: InputAutocompleteMobileProps,
        ref,
    ) => {
        const [open, setOpen] = useState(false);
        const frozenValue = useRef<string>('');
        const searchInputRef = useRef<HTMLInputElement>(null);
        const targetRef = useRef<HTMLDivElement>(null);

        const restorePrevValue = () => onInput?.(frozenValue.current, OnInputReason.Close);

        const setModalVisibility = (isOpen: boolean) => {
            if (isOpen) {
                frozenValue.current = value || '';
            }

            if (openProp === undefined) {
                setOpen(isOpen);
            }

            onOpen?.({ open: isOpen, name });
        };

        const handleOpen: SelectMobileProps['onOpen'] = (payload) => {
            setModalVisibility(Boolean(payload.open));
        };

        const handleOptionsListTouchMove = useMemo(
            () =>
                throttle(() => {
                    const input = searchInputRef.current;

                    if (input && document.activeElement === input) {
                        input.blur();
                    }
                }, 300),
            [],
        );

        const handleApply = () => {
            setModalVisibility(false);
            onApply?.();
        };

        const handleCancel = () => {
            setModalVisibility(false);
            restorePrevValue();
            onCancel?.();
        };

        const handleExiting = (node: HTMLElement) => {
            targetRef.current?.focus();
            transitionProps?.onExiting?.(node);
        };

        const isOpen = Boolean(open || openProp);

        const Component = isBottomSheet ? SelectMobile : SelectModalMobile;

        const componentProps:
            | ModalSelectMobileProps['modalProps']
            | BottomSheetSelectMobileProps['bottomSheetProps'] = {
            title: title || label || placeholder,
            onClose: restorePrevValue,
            transitionProps: {
                ...transitionProps,
                onExiting: handleExiting,
            },
            [isBottomSheet ? 'containerProps' : 'componentDivProps']: {
                onTouchMove: handleOptionsListTouchMove,
            },
        };

        const clear = inputProps?.clear ?? false;

        return (
            <Component
                Field={AutocompleteMobileField}
                {...restProps}
                {...(isBottomSheet
                    ? { bottomSheetProps: componentProps }
                    : {
                          modalProps: componentProps,
                          modalHeaderProps: { title },
                      })}
                dataTestId={dataTestId}
                useWithApplyHook={false}
                showSearch={true}
                searchProps={{
                    value,
                    filterFn: searchFilterStub,
                    componentProps: {
                        leftAddons: null,
                        placeholder,
                        ...inputProps,
                        className: cn(styles.input, inputProps?.className),
                        clear,
                        ref: mergeRefs([searchInputRef, inputProps?.ref as Ref<HTMLInputElement>]),
                        onChange: (_, payload) => onInput?.(payload.value, OnInputReason.Change),
                    },
                }}
                Search={Input}
                ref={mergeRefs([targetRef, ref])}
                open={isOpen}
                onOpen={handleOpen}
                Arrow={Arrow}
                placeholder={placeholder}
                label={label}
                size={size}
                name={name}
                multiple={multiple}
                optionsListProps={{
                    footer: (
                        <Footer
                            showClear={true}
                            handleClear={handleCancel}
                            handleApply={handleApply}
                            clearText='Отмена'
                            applyText='Продолжить'
                            dataTestId={dataTestId}
                        />
                    ),
                    ...(restProps.optionsListProps as AnyObject),
                }}
                fieldProps={{
                    value: isOpen ? frozenValue.current : value,
                    clear,
                    onClear: clear ? inputProps?.onClear : undefined,
                    success,
                    ...(restProps.fieldProps as AnyObject),
                }}
            />
        );
    },
);
