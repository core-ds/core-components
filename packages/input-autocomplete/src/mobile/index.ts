export { InputAutocompleteMobile, InputAutocompleteMobileProps } from '../Component.mobile';
export {
    InputAutocompleteModalMobile,
    InputAutocompleteModalMobileProps,
} from '../Component.modal.mobile';
