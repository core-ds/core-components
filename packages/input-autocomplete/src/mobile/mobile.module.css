@import '../../../themes/src/default.css';

.input {
    width: auto;
}
