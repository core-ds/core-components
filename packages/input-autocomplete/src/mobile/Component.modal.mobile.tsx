import React from 'react';

import { InputAutocompleteMobileProps } from '../types';

import { InputAutocompleteMobile } from './Component.mobile';

export const InputAutocompleteModalMobile = React.forwardRef<
    HTMLDivElement,
    InputAutocompleteMobileProps
>((props, ref) => <InputAutocompleteMobile {...props} ref={ref} isBottomSheet={false} />);
