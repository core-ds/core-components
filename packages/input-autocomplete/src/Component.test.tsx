import React, { useState } from 'react';
import { fireEvent, render, waitFor, waitForElementToBeRemoved } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { InputAutocompleteMobile, InputAutocompleteMobileProps } from './mobile';

const dataTestId = 'test-id';

const options = [
    { key: '1', content: 'Neptunium' },
    { key: '2', content: 'Plutonium' },
];

const noop = () => {};

const InputAutocompleteMobileWrapper = (props: Partial<InputAutocompleteMobileProps>) => {
    const [value, setValue] = useState('');
    const [open, setOpen] = useState(props.open === undefined ? true : props.open);

    const handleOpen: InputAutocompleteMobileProps['onOpen'] = ({ open: isOpen }) => {
        setOpen(Boolean(isOpen));
    };

    const handleInput: InputAutocompleteMobileProps['onInput'] = (event, payload) => {
        setValue(payload.value);
        props?.onInput?.(event, payload);
    };

    return (
        <InputAutocompleteMobile
            open={open}
            onOpen={handleOpen}
            options={options}
            value={value}
            onChange={noop}
            {...props}
            onInput={handleInput}
            transitionProps={{
                timeout: 0,
                ...props.transitionProps,
            }}
        />
    );
};

describe('InputAutocompleteMobile', () => {
    describe('Snapshots tests', () => {
        Object.defineProperty(window, 'matchMedia', {
            writable: true,
            value: jest.fn().mockImplementation((query) => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(),
                removeListener: jest.fn(),
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            })),
        });

        it('should match snapshot in closed state', () => {
            const { baseElement } = render(<InputAutocompleteMobileWrapper open={false} />);

            expect(baseElement).toMatchSnapshot();
        });

        it('should match snapshot in open state', () => {
            const { baseElement } = render(<InputAutocompleteMobileWrapper />);

            expect(baseElement).toMatchSnapshot();
        });
    });

    describe('Props tests', () => {
        it('should set `data-test-id` attribute', () => {
            const { getByTestId } = render(
                <InputAutocompleteMobileWrapper dataTestId={dataTestId} />,
            );

            expect(getByTestId(dataTestId).tagName).toBe('DIV');
            expect(getByTestId(dataTestId).getAttribute('role')).toBe('combobox');
        });

        it('should be open, if open is `true`', () => {
            const { getByRole } = render(<InputAutocompleteMobileWrapper open={true} />);

            expect(getByRole('dialog')).toBeInTheDocument();
        });

        it('should render title', () => {
            const label = 'Label';

            const { getByRole } = render(
                <InputAutocompleteMobileWrapper label={label} open={true} />,
            );

            expect(getByRole('dialog').querySelector('div[class^="title"]')).toBeInTheDocument();
        });

        it('should display title instead of label in bottom-sheet', () => {
            const label = 'Label';
            const title = 'Title';

            const { getByRole } = render(
                <InputAutocompleteMobileWrapper
                    title={title}
                    label={label}
                    open={true}
                    isBottomSheet={true}
                />,
            );

            expect(getByRole('dialog').querySelector('div[class^="header"]')?.textContent).toBe(
                title,
            );
        });

        it('should display title instead of label in modal', () => {
            const label = 'Label';
            const title = 'Title';

            const { getByRole } = render(
                <InputAutocompleteMobileWrapper
                    title={title}
                    label={label}
                    open={true}
                    isBottomSheet={false}
                />,
            );

            expect(getByRole('dialog').querySelector('div[class^="header"]')?.textContent).toBe(
                title,
            );
        });
    });

    describe('Interactions tests', () => {
        it('should close on cancel button click', async () => {
            const closeButtonId = dataTestId + '-clear';
            const handleEnter = jest.fn();

            const { getByTestId, getByRole, queryByTestId } = render(
                <InputAutocompleteMobileWrapper
                    dataTestId={dataTestId}
                    transitionProps={{
                        onEnter: handleEnter,
                    }}
                />,
            );

            await waitFor(() => expect(handleEnter).toHaveBeenCalled(), { timeout: 5000 });

            fireEvent.click(getByTestId(closeButtonId));

            await waitForElementToBeRemoved(() => getByRole('dialog'));

            expect(queryByTestId(closeButtonId)).not.toBeInTheDocument();
        });

        it('should close on continue button click', async () => {
            const continueButtonId = dataTestId + '-apply';
            const handleChange = jest.fn();

            const { getByTestId, getByRole, queryByTestId } = render(
                <InputAutocompleteMobileWrapper dataTestId={dataTestId} onChange={handleChange} />,
            );

            fireEvent.click(getByTestId(continueButtonId));

            await waitForElementToBeRemoved(() => getByRole('dialog'));

            expect(queryByTestId(continueButtonId)).not.toBeInTheDocument();
        });
    });

    describe('Callback tests', () => {
        it('should call onChange', async () => {
            const cb = jest.fn();
            const { findByText } = render(<InputAutocompleteMobileWrapper onChange={cb} />);

            const option = await findByText(options[0].content);
            fireEvent.click(option);
            expect(cb).toBeCalledTimes(1);
        });

        it('should call onApply', async () => {
            const cb = jest.fn();
            const { getByTestId } = render(
                <InputAutocompleteMobileWrapper onApply={cb} dataTestId={dataTestId} />,
            );

            fireEvent.click(getByTestId(dataTestId + '-apply'));
            expect(cb).toBeCalledTimes(1);
        });

        it('should call onCancel', async () => {
            const cb = jest.fn();
            const { getByTestId } = render(
                <InputAutocompleteMobileWrapper onCancel={cb} dataTestId={dataTestId} />,
            );

            fireEvent.click(getByTestId(dataTestId + '-clear'));
            expect(cb).toBeCalledTimes(1);
        });

        it('should call onBlur', async () => {
            const onBlur = jest.fn();
            const onExited = jest.fn();
            const { findByText } = render(
                <InputAutocompleteMobileWrapper onBlur={onBlur} transitionProps={{ onExited }} />,
            );

            const option = await findByText(options[0].content);

            fireEvent.click(option);
            await waitFor(() => expect(onExited).toBeCalled(), {
                timeout: 4000,
            });
            await userEvent.tab();

            expect(onBlur).toBeCalled();
        });

        it('should restore prev value when click cancel ', async () => {
            const cb = jest.fn();
            const { queryByRole, getByTestId } = render(
                <InputAutocompleteMobileWrapper onInput={cb} dataTestId={dataTestId} />,
            );

            const input = queryByRole('textbox') as HTMLInputElement;

            fireEvent.change(input, { target: { value: '123' } });
            fireEvent.click(getByTestId(dataTestId + '-clear'));

            expect(cb).toHaveBeenNthCalledWith(2, expect.any(Object), { value: '' });
        });

        it('should restore prev value when click close button ', () => {
            const cb = jest.fn();
            const { queryByRole, getByTestId } = render(
                <InputAutocompleteMobileWrapper onInput={cb} dataTestId={dataTestId} />,
            );

            const input = queryByRole('textbox') as HTMLInputElement;

            fireEvent.change(input, { target: { value: '123' } });
            fireEvent.click(getByTestId(dataTestId + '-bottom-sheet-header-closer'));

            expect(cb).toHaveBeenNthCalledWith(2, expect.any(Object), { value: '' });
        });

        it('should apply new value when click continue ', () => {
            const cb = jest.fn();
            const { queryByRole, getByTestId } = render(
                <InputAutocompleteMobileWrapper onInput={cb} dataTestId={dataTestId} />,
            );

            const input = queryByRole('textbox') as HTMLInputElement;

            fireEvent.change(input, { target: { value: '123' } });
            fireEvent.click(getByTestId(dataTestId + '-apply'));

            expect(cb).toBeCalledTimes(1);
            expect(cb).toBeCalledWith(expect.any(Object), { value: '123' });
        });
    });

    describe('Render tests', () => {
        it('should unmount without errors', () => {
            const { unmount } = render(<InputAutocompleteMobileWrapper />);

            expect(unmount).not.toThrowError();
        });
    });
});
