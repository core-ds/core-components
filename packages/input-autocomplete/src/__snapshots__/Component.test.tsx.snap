// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAutocompleteMobile Snapshots tests should match snapshot in closed state 1`] = `
<body>
  <div>
    <div
      aria-expanded="false"
      aria-haspopup="listbox"
      class="component"
      role="combobox"
      tabindex="-1"
    >
      <div
        class="component"
      >
        <div
          class="component component size-48 size-48 block"
        >
          <div
            aria-controls="downshift-:r0:-menu"
            aria-labelledby="downshift-:r0:-label"
            class="field inner inner inner inner"
            id="downshift-:r0:-input"
            tabindex="0"
          >
            <div
              class="inputWrapper"
            >
              <div
                class="input"
              >
                <div
                  class="contentWrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`InputAutocompleteMobile Snapshots tests should match snapshot in open state 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div>
    <div
      aria-expanded="true"
      aria-haspopup="listbox"
      class="component"
      role="combobox"
      tabindex="-1"
    >
      <div
        class="component"
      >
        <div
          class="component component size-48 size-48 block"
        >
          <div
            aria-controls="downshift-:r1:-menu"
            aria-labelledby="downshift-:r1:-label"
            class="field inner inner inner inner"
            id="downshift-:r1:-input"
          >
            <div
              class="inputWrapper"
            >
              <div
                class="input"
              >
                <div
                  class="contentWrapper"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    alfa-portal-container=""
  >
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
    <div
      data-focus-lock-disabled="false"
    >
      <div
        style="opacity: 1; transition: opacity 500ms ease-in-out; position: relative; z-index: 100;"
      >
        <div
          aria-hidden="true"
          class="backdrop disabledPointerEvents appear appearActive"
        />
      </div>
      <div
        class="wrapper"
        role="dialog"
        style="z-index: 100;"
        tabindex="-1"
      >
        <div
          class="component modal appear appearActive"
        >
          <div
            class="content"
          >
            <div
              class="wrapper"
            >
              <div
                class="component sheet"
                style="max-height: 744px; height: 744px;"
              >
                <div
                  class="marker defaultMarker"
                />
                <div
                  class="scrollableContainer sheetContainer"
                >
                  <div
                    class="header headerWrapper sticky"
                  >
                    <div
                      class="mainLine"
                    >
                      <div
                        class="addonsWrapper rightAddons"
                      >
                        <div
                          class="addon closer"
                        >
                          <div
                            class="closer"
                          >
                            <button
                              aria-label="закрыть"
                              class="component text size-48 hug size-48 component text component text cc-icon-button button primary component border-32"
                              type="button"
                            >
                              <span
                                class="label stretchText"
                              >
                                <span
                                  class="iconWrapper size-32 center"
                                >
                                  <svg
                                    class="icon"
                                    fill="currentColor"
                                    focusable="false"
                                    height="24"
                                    role="img"
                                    viewBox="0 0 24 24"
                                    width="24"
                                  >
                                    <path
                                      d="M10.586 12L6 16.5 7.5 18l4.5-4.586L16.5 18l1.5-1.5-4.586-4.5L18 7.5 16.5 6 12 10.586 7.5 6 6 7.5l4.586 4.5z"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                    <div
                      class="bottomAddons"
                    >
                      <div
                        class="component component component input search size-48 size-48"
                      >
                        <div
                          class="inner inner inner inner"
                        >
                          <div
                            class="inputWrapper"
                          >
                            <div
                              class="input"
                            >
                              <input
                                class="input input"
                                placeholder=""
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="content sheetContent noFooter"
                  >
                    <div
                      aria-labelledby="downshift-:r1:-label"
                      class="optionsListWrapper"
                      id="downshift-:r1:-menu"
                      role="listbox"
                    >
                      <div
                        class="optionsList size-48 optionsList"
                      >
                        <div
                          class="component component scrollable scrollbar"
                          data-simplebar="init"
                          style="min-width: 0;"
                        >
                          <div
                            class="wrapper"
                          >
                            <div
                              class="heightAutoObserverWrapper"
                            >
                              <div
                                class="heightAutoObserver"
                              />
                            </div>
                            <div
                              class="mask mask"
                              style=""
                            >
                              <div
                                class="offset"
                                style="right: 0px; bottom: 0px;"
                              >
                                <div
                                  aria-label="scrollable content"
                                  class="contentWrapper"
                                  role="region"
                                  style="height: auto; overflow-x: hidden; overflow-y: hidden;"
                                  tabindex="0"
                                >
                                  <div
                                    class="content"
                                    style=""
                                  >
                                    <div
                                      aria-disabled="false"
                                      aria-selected="false"
                                      class="option size-56 option"
                                      id="downshift-:r1:-item-0"
                                      role="option"
                                    >
                                      <div
                                        class="checkmark"
                                      >
                                        <svg
                                          class="displayIcon"
                                          fill="currentColor"
                                          focusable="false"
                                          height="24"
                                          role="img"
                                          viewBox="0 0 24 24"
                                          width="24"
                                        >
                                          <path
                                            clip-rule="evenodd"
                                            d="M20 7.519L9.903 17 5 12.327l1.448-1.503 3.47 3.308L18.569 6 20 7.519z"
                                            fill-rule="evenodd"
                                          />
                                        </svg>
                                      </div>
                                      <div
                                        class="content textContent"
                                      >
                                        Neptunium
                                      </div>
                                      <div
                                        class="checkmark"
                                      >
                                        <svg
                                          class="displayIcon"
                                          fill="currentColor"
                                          focusable="false"
                                          height="24"
                                          role="img"
                                          viewBox="0 0 24 24"
                                          width="24"
                                        >
                                          <path
                                            clip-rule="evenodd"
                                            d="M20 7.519L9.903 17 5 12.327l1.448-1.503 3.47 3.308L18.569 6 20 7.519z"
                                            fill-rule="evenodd"
                                          />
                                        </svg>
                                      </div>
                                    </div>
                                    <div
                                      aria-disabled="false"
                                      aria-selected="false"
                                      class="option size-56 option"
                                      id="downshift-:r1:-item-1"
                                      role="option"
                                    >
                                      <div
                                        class="checkmark"
                                      >
                                        <svg
                                          class="displayIcon"
                                          fill="currentColor"
                                          focusable="false"
                                          height="24"
                                          role="img"
                                          viewBox="0 0 24 24"
                                          width="24"
                                        >
                                          <path
                                            clip-rule="evenodd"
                                            d="M20 7.519L9.903 17 5 12.327l1.448-1.503 3.47 3.308L18.569 6 20 7.519z"
                                            fill-rule="evenodd"
                                          />
                                        </svg>
                                      </div>
                                      <div
                                        class="content textContent"
                                      >
                                        Plutonium
                                      </div>
                                      <div
                                        class="checkmark"
                                      >
                                        <svg
                                          class="displayIcon"
                                          fill="currentColor"
                                          focusable="false"
                                          height="24"
                                          role="img"
                                          viewBox="0 0 24 24"
                                          width="24"
                                        >
                                          <path
                                            clip-rule="evenodd"
                                            d="M20 7.519L9.903 17 5 12.327l1.448-1.503 3.47 3.308L18.569 6 20 7.519z"
                                            fill-rule="evenodd"
                                          />
                                        </svg>
                                      </div>
                                    </div>
                                  </div>
                                </div>
                              </div>
                            </div>
                            <div
                              class="placeholder"
                              style="width: 0px; height: 0px;"
                            />
                          </div>
                          <div
                            class="track horizontal"
                            style="visibility: hidden;"
                          >
                            <div
                              class="scrollbar"
                              style="width: 0px; display: none;"
                            />
                          </div>
                          <div
                            class="track vertical"
                            style="visibility: hidden;"
                          >
                            <div
                              class="scrollbar"
                              style="height: 0px; display: none;"
                            />
                          </div>
                        </div>
                        <div
                          class="optionsListFooter"
                        >
                          <div
                            class="footer"
                            tabindex="0"
                          >
                            <button
                              class="component secondary size-56 hug size-56 component secondary component secondary button"
                              type="button"
                            >
                              <span
                                class="label stretchText"
                              >
                                Отмена
                              </span>
                            </button>
                            <button
                              class="component primary size-56 hug size-56 component primary component primary button"
                              type="button"
                            >
                              <span
                                class="label stretchText"
                              >
                                Продолжить
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      data-focus-guard="true"
      style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
      tabindex="0"
    />
  </div>
</body>
`;
